src/main/java/seedu/address/logic/parser/DeleteVisitsCommandParser.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+
+import java.time.LocalDate;
+import java.util.stream.Stream;
+
+import seedu.address.logic.commands.DeleteVisitsCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new DeleteVisitsCommand object
+ */
+public class DeleteVisitsCommandParser implements Parser<DeleteVisitsCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DeleteVisitsCommand
+     * and returns a DeleteVisitsCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteVisitsCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_DATE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_DATE) || argMultimap.getPreamble().isEmpty()) {
```

Comment:
argMultimap.getPreamble().isEmpty() should be !argMultimap.getPreamble().isEmpty(), otherwise the code fails.

Reply:
Only this part causes errors; the other comments are queries/small formatting errors.

Reply:
ohh i get it, so only parsing in the lower half of args without the command word